src/frontend/js/pages/TeacherCourseDashboardLoader/CourseRunList/utils.tsx

```diff
@@ -0,0 +1,78 @@
+import { FormattedMessage, IntlShape, defineMessages } from 'react-intl';
+import { capitalize } from 'lodash-es';
+import { Button } from '@openfun/cunningham-react';
+import { IconTypeEnum } from 'components/Icon';
+import { CourseStateTextEnum } from 'types';
+import { CourseRun } from 'types/Joanie';
+import CourseRunListCell from './CourseRunListCell';
+
+export const messages = defineMessages({
+  dataCourseRunPeriod: {
+    defaultMessage: 'from {from} to {to}',
+    description: 'Message displayed in course run datagrid for course run period',
+    id: 'components.TeacherCourseDashboardLoader.dataCourseRunPeriod',
+  },
+  dataCourseRunLink: {
+    defaultMessage: 'go to classroom',
+    description: 'Message displayed in course run datagrid for course run link',
+    id: 'components.TeacherCourseDashboardLoader.dataCourseRunLink',
+  },
+});
+
+export const buildCourseRunData = (intl: IntlShape, courseRuns: CourseRun[]) => {
+  const CourseStateIconMap: Record<CourseStateTextEnum, IconTypeEnum> = {
+    [CourseStateTextEnum.CLOSING_ON]: IconTypeEnum.MORE,
+    [CourseStateTextEnum.STARTING_ON]: IconTypeEnum.CHECK_ROUNDED,
+    [CourseStateTextEnum.ENROLLMENT_CLOSED]: IconTypeEnum.MORE,
+    [CourseStateTextEnum.ON_GOING]: IconTypeEnum.MORE,
+    [CourseStateTextEnum.ARCHIVED]: IconTypeEnum.ARCHIVE,
+    [CourseStateTextEnum.TO_BE_SCHEDULED]: IconTypeEnum.STOPWATCH,
+  };
+
+  return courseRuns.map((courseRun: CourseRun) => {
+    return {
```

Comment:
Little tips for the style 🤷‍♂️

```suggestion
  return courseRuns.map((courseRun: CourseRun) => ({
```